internal/cmd/rc.go

```diff
@@ -139,6 +139,7 @@ func (rc *RC) Load(previousEnv Env) (newEnv Env, err error) {
 	direnv := config.SelfPath
 	newEnv = previousEnv.Copy()
 	newEnv[DIRENV_WATCHES] = rc.times.Marshal()
+	newEnv[DIRENV_IGNORE_KEYS] = strings.Join(config.EnvIgnoreKeys, ",")
```

Comment:
Passing through environment variable avoids the need to read configuration file in at least `direnv apply_dump`. 

As a (wanted) side effect, it also adds appending to the list in `.envrc` for example in some `direnv_load ... direnv dump` scenarios. Although I think we can leave that undocumented if this still changes.